docker/docker_image.go

```diff
@@ -103,3 +107,50 @@ func GetRepositoryTags(ctx context.Context, sys *types.SystemContext, ref types.
 	}
 	return tags, nil
 }
+
+// GetDigest returns the image's digest. (This is a
+// backward-compatible shim method which calls the module-level
+// GetDigest)
+func (i *Image) GetDigest(ctx context.Context) (*digest.Digest, error) {
+	return GetDigest(ctx, i.src.c.sys, i.src.logicalRef)
+}
+
+// GetDigest returns the image's digest
```

Comment:
Absolutely non-blocking: Also maybe
```suggestion
// GetDigest returns the image's digest
// Use this to optimize and avoid use of an ImageSource based on the returned digest;
// if you are going to use an ImageSource anyway, it’s more efficient to create it first
// and compute the digest from the value returned by GetManifest.
```
to explain the trade-offs.